java/squeek/veganoption/content/registry/DescriptionRegistry.java

```diff
@@ -37,38 +36,62 @@ public static void registerAllDescriptions()
 	public static boolean tryRegisterItem(Item item)
 	{
 		boolean didRegister = false;
-		if (hasUsageText(item) && !itemsWithUsageDescriptions.contains(item))
+
+		if (damageableItemsWithUniqueDescriptions.contains(item))
+		{
+			ItemStack stack = new ItemStack(item);
+			for (int damage = 1; damage <= stack.getMaxDamage(); damage++)
+			{
+				stack.setDamageValue(damage);
+				boolean registeredStack = tryRegisterItemStack(stack);
+				if (registeredStack)
+					didRegister = true;
+			}
+		}
+		didRegister = tryRegisterItemStack(new ItemStack(item)) || didRegister;
+		return didRegister;
+	}
+
+	public static boolean tryRegisterItemStack(ItemStack stack)
+	{
+		boolean didRegister = false;
+		if (hasUsageText(stack) && itemsWithUsageDescriptions.stream().noneMatch(i -> i.getItem() == stack.getItem()))
 		{
-			itemsWithUsageDescriptions.add(item);
+			itemsWithUsageDescriptions.add(stack);
+			registeredLangKeys.add(stack.getDescriptionId() + ".vowiki.usage");
 			didRegister = true;
 		}
-		if (hasCraftingText(item) && !itemsWithCraftingDescriptions.contains(item))
+		if (hasCraftingText(stack) && itemsWithCraftingDescriptions.stream().noneMatch(i -> i.getItem() == stack.getItem()))
 		{
-			itemsWithCraftingDescriptions.add(item);
+			itemsWithCraftingDescriptions.add(stack);
+			registeredLangKeys.add(stack.getDescriptionId() + ".vowiki.crafting");
 			didRegister = true;
 		}
 		return didRegister;
 	}
 
-	public static boolean hasUsageText(Item item)
+	private static boolean hasUsageText(ItemStack stack)
 	{
-		return LangHelper.existsRaw(item.getDescriptionId() + ".vowiki.usage") || !RelationshipRegistry.getChildren(item).isEmpty();
+		String key = stack.getDescriptionId() + ".vowiki.usage";
+		return !registeredLangKeys.contains(key) && (LangHelper.existsRaw(key) || !RelationshipRegistry.getChildren(stack.getItem()).isEmpty());
```

Comment:
The registeredLangKeys checks should be taken out of the hasUsage/CraftingText methods as it being registered already doesn't mean, logically, that it doesn't have usage/crafting text. Perhaps a new set of canUsage/CraftingBeRegistered methods that does the registeredLangKeys check and the itemsWithDescriptions check